src/java/org/apache/cassandra/repair/LocalSyncTask.java

```diff
@@ -75,7 +75,7 @@ StreamPlan createStreamPlan()
     {
         InetAddressAndPort remote =  nodePair.peer;
 
-        StreamPlan plan = new StreamPlan(StreamOperation.REPAIR, 1, false, pendingRepair, previewKind)
+        StreamPlan plan = new StreamPlan(desc.determanisticId(), StreamOperation.REPAIR, 1, false, pendingRepair, previewKind)
```

Comment:
the idea is that streaming done by repair will use the RepairJob id (as defined by CASSANDRA-15399), so if a repair exists with the same planId, then this stream can map to the repair without any networking changes